pkg/apis/core/v1/defaults.go

```diff
@@ -133,32 +132,32 @@ func SetDefaults_Service(obj *v1.Service) {
 		obj.Spec.ExternalTrafficPolicy = v1.ServiceExternalTrafficPolicyTypeCluster
 	}
 
-	// if dualstack feature gate is on then we need to default
-	// Spec.IPFamily correctly. This is to cover the case
-	// when an existing cluster have been converted to dualstack
-	// i.e. it already contain services with Spec.IPFamily==nil
-	if utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) &&
```

Comment:
if the gate is part of dependency chain then, go client will have to check the gate status before making any assumption about the IPFamilies/Policy etc. Removing the gate means go-clients (or any other for that matter) can work with Service object knowing that policy/families and number of cluster IPs will be correct irrespective of what cluster they are connected to. 

if a new client (with families) doing if `svc.IPFamilies == nil` then it can easily know that the api-server connected is using older APIs.

This is a hill i am willing to die on :-) so let us talk it out.

Reply:
If clients are allowed to make use of those fields without knowing whether the feature gate is enabled or disabled, then those fields are effectively GA...

Reply:
defaulting is now gated..